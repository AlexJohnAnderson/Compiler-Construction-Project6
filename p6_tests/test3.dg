int a() { return 1; }
int b() { return 2; }
int main(){
fn ()-> int ptr;
if(mayhem > 2) {
    ptr = a;
    } else {
        ptr = b;
    }
    ptr();
}