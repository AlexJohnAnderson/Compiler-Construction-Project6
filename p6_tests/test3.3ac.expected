[BEGIN GLOBALS]
a
b
main
[END GLOBALS]
[BEGIN a LOCALS]
[END a LOCALS]
fun_a:      enter a
            setret 1
            goto lbl_0
lbl_0:      leave a
[BEGIN b LOCALS]
[END b LOCALS]
fun_b:      enter b
            setret 2
            goto lbl_1
lbl_1:      leave b
[BEGIN main LOCALS]
ptr (local var of 8 bytes)
tmp0 (tmp var of 8 bytes)
tmp1 (tmp var of 8 bytes)
tmp2 (tmp var of 8 bytes)
[END main LOCALS]
main:       enter main
            MAYHEM [tmp0]
            [tmp1] := [tmp0] GT64 2
            IFZ [tmp1] GOTO lbl_3
            [ptr] := [a]
            goto lbl_4
lbl_3:      nop
            [ptr] := [b]
lbl_4:      nop
            call ptr
lbl_2:      leave main
